### ISSUE TITLE
Zelda TOTK - Shader compilation-like stutter even after shader is built

### ISSUE THREAD

**Issue by sallliisa** [yuzu-emu/yuzu#13094]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1721 (I've played since I think mainline 1701 and the issue still persists)

### Description of Issue

Running on an i3 10105f + AMD RX 6600 (tested all drivers through 23.11.1 to 24.1.1, observed no changes at all). Async shader compilation + usage of disk-cached Vulkan pipeline is active. Hell, everything is active.

Have tried deleting shader cache, both on driver-level and on Yuzu, issue still persists. Have tried game version 1.0.0, 1.2.0, and 1.2.1, issue still persists (each game version is tested with a fresh shader cache). I have tried running the game with and without mods and issue still persists.  Have tried running it off of an NVMe SSD, issue still persists.

Having the same issue with Ryujinx ONLY if PPTC is inactive, otherwise it runs smooth and flawless.

[Here's a recording of the issue + I left the log console open (with "Enable Shader Feedback" and "Enable FS Access Log" on)](https://youtu.be/exfdAwSfn5M)

Here's my Yuzu setting for the game
![image](https://github.com/yuzu-emu/yuzu/assets/89846756/4b3cfa7e-9316-4fba-929a-360508d91977)
![image](https://github.com/yuzu-emu/yuzu/assets/89846756/d7adb493-9606-4151-930e-e57114442fcc)

I have tried enabling "Force maximum clocks" but the issue still persists.

### Expected Behavior

Minimal stutter after shader is compiled

### Reproduction Steps

Run the game -> stutters every time a new "effect" is being shown on screen

Restart the game (doesnt even have to restart Yuzu) -> still stutters

### Log File

[yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/14351135/yuzu_log.txt.old.txt)

### System Configuration

CPU: i3 10105f
GPU/Driver: AMD RX 6600/23.11.1, 23.12.1, 24.1.1
RAM: 16GB
OS: Windows 11 22631

**Comment by german77** [comment]
totk creates infinite shaders. it will always stutter

**Comment by goldenx86** [comment]
That's also ASTC encoding.

**Comment by sallliisa** [comment]
> That's also ASTC encoding.

Do you mean the stutter is caused by ASTC textures? Sorry I forgot to mention in the post that I've tried changing ASTC decoding to async CPU and tried all recompression method, and still no dice. Using async CPU ASTC decoding introduces some graphical glitches so I settled on using GPU instead. I'll add it to the post though, thanks!

**Comment by sallliisa** [comment]
> totk creates infinite shaders. it will always stutter

Do you mean that it's a fault at Yuzu's part of constantly invalidating shader caches or that there's so many shaders in TOTK that the stutter I experienced actually comes from a conventional shadercomp stutter? I don't think the latter is what's happening since as what the log console has shown in the recording that I provided, no pipelines are being created at the time of stutter

**Comment by Squall-Leonhart** [comment]
theres a pipeline state for every location with every combination of gear at every time of day or weather type

**Comment by sallliisa** [comment]
> theres a pipeline state for every location with every combination of gear at every time of day or weather type

Yes, but as shown in the recording I loaded the exact same save; does not stutter if I "soft-reloaded" the game via in-game menus, stutters if I restarted the emulation
